libdnf5/module/module_sack.cpp

```diff
@@ -777,14 +778,90 @@ bool ModuleSack::Impl::enable(const std::string & name, const std::string & stre
+    const ModuleQuery & module_query) {
+    std::map<std::string, std::map<std::string, std::vector<int>>> module_dict;
+    for (const auto & module_item : module_query.list()) {
+        module_dict[module_item.get_name()][module_item.get_stream()].push_back(module_item.get_id().id);
+    }
+    return module_dict;
+}
+
+
+std::vector<std::string> ModuleSack::Impl::prune_module_dict(
+    std::map<std::string, std::map<std::string, std::vector<int>>> & module_dict) {
+    // Vector of module names with multiple streams to enable
+    std::vector<std::string> multiple_stream_modules;
+
+    for (auto & module_dict_iter : module_dict) {
+        auto name = module_dict_iter.first;
+        auto & stream_dict = module_dict_iter.second;
+        auto module_status = module_db->get_status(name);
+
+        // Multiple streams match the requested spec
+        if (stream_dict.size() > 1) {
+            // Get stream that is either enabled (for ENABLED module), or default (otherwise)
+            std::string enabled_or_default_stream;
+            if (module_status == ModuleStatus::ENABLED) {
+                enabled_or_default_stream = module_db->get_enabled_stream(name);
+            } else {
+                enabled_or_default_stream = module_sack->get_default_stream(name);
+            }
+
+            // Module doesn't have any enabled nor default stream
+            if (enabled_or_default_stream.empty()) {
+                multiple_stream_modules.emplace_back(name);
+                continue;
+            }
+
+            // The enabled or default stream is not one of the possible streams
+            if (stream_dict.find(enabled_or_default_stream) == stream_dict.end()) {
+                multiple_stream_modules.emplace_back(name);
+                continue;
+            }
+
+            // Remove all streams except for the enabled_or_default_stream
+            for (auto iter = stream_dict.begin(); iter != stream_dict.end();) {
+                if (iter->first != enabled_or_default_stream) {
+                    stream_dict.erase(iter++);
+                } else {
+                    ++iter;
+                }
+            }
+        }
+    }
+    return multiple_stream_modules;
+}
+
+
 bool ModuleSack::Impl::enable(const std::string & module_spec, bool count) {
     module_db->initialize();
 
+    // For the given module_spec, create a dict { name : {stream : [solvable id] }
+    auto module_dict = create_module_dict(module_spec_to_query(base, module_spec));
+    // Keep only enabled or default streams if possible
+    auto multiple_stream_modules = prune_module_dict(module_dict);
+    // If there are any modules with multiple streams to be enabled, throw an error
+    if (!multiple_stream_modules.empty()) {
+        throw EnableMultipleStreamsError(
+            M_("Cannot enable multiple streams of one module at the same time. Affected modules: {}"),
+            utils::string::join(multiple_stream_modules, ", "));
```

Comment:
I believe that in some aspects DNF4 provided a better message.
```
sudo dnf module enable '*' '*m*'
[sudo] password for jmracek: 
Last metadata expiration check: 2:04:24 ago on Wed 17 Jan 2024 12:17:10 PM CET.
Argument '*' matches 5 streams ('10.5', '10.6', '10.7', '10.8', '10.9') of module 'mariadb', but none of the streams are enabled or default
Unable to resolve argument *
Argument '*m*' matches 5 streams ('10.5', '10.6', '10.7', '10.8', '10.9') of module 'mariadb', but none of the streams are enabled or default
Unable to resolve argument *m*
Error: Problems in request:
broken groups or modules: *, *m*
```
Your code is better because it provides information about all modules with multiple stream for one spec.

```
sudo dnf5 module enable '*' '*m*'
Updating and loading repositories:
Repositories loaded.
Cannot enable multiple streams of one module at the same time. Affected modules: avocado, avocado-vt, cri-o, mariadb, nextcloud, nginx, perl, perl-DBD-SQLite, perl-libwww-perl, postgresql, swig
```
The new code does not show information about available streams available streams. It means that user has to run additional command to list all streams for spec.

The critical problem - DNF5 does not show error message for all arguments but only for the first one.
Important problem - if I am not mistaken (I am guessing, following error catch path is quite demanding) - problem is not logged into event log. It means API users will have a difficulty to understand the issue. I am investigation it and I will add additional information in future.